syntax = "proto2";
package units.proto;

import "games/actions/proto/plan.proto";
import "games/actions/proto/strategy.proto";
import "games/geography/proto/geography.proto";
import "games/market/proto/goods.proto";
import "util/proto/object_id.proto";

message Unit {
  // The kind corresponds to the kind of a unit template.
  optional util.proto.ObjectId unit_id = 1;
  optional geography.proto.Location location = 2;
  optional market.proto.Container resources = 3;
  optional actions.proto.Plan plan = 4;
  optional actions.proto.Strategy strategy = 5;
  optional util.proto.ObjectId faction_id = 6;
}
